File: Pemrograman Berbasis Objek/Praktikum 1/3/Kompleks.cpp
#include "Kompleks.h"
#include <cstdio>

int Kompleks::n_kompleks = 0;

//int real;
//int imaginer;

Kompleks::Kompleks() {
	this->real = 0;
	this->imaginer = 0;
	n_kompleks++;
}
	
Kompleks::Kompleks(int real, int imaginer) {
	this->real = real;
	this->imaginer = imaginer;
	n_kompleks++;
}

int Kompleks::GetReal() const {
	return this->real;
}

int Kompleks::GetImaginer() const {
	return this->imaginer;
}

void Kompleks::SetReal(int x) {
	this->real = x;
}

void Kompleks::SetImaginer(int x) {
	this->imaginer = x;
}

// (a + bi) + (c + di) = (a+c) + (b+d)i
Kompleks operator+ (const Kompleks& a, const Kompleks& b) {
	Kompleks c;
	c.real = a.real + b.real;
	c.imaginer = a.imaginer + b.imaginer;
	return c;
	//turn Kompleks::Kompleks(a.real+b.real, a.imaginer()+b.imaginer());
}

// (a + bi) − (c + di) = (a−c) + (b−d)i
Kompleks operator- (const Kompleks& a, const Kompleks& b) {
	Kompleks c;
	c.real = a.real - b.real;
	c.imaginer = a.imaginer - b.imaginer;
	return c;
	//turn Kompleks::Kompleks(a.real-b.real, a.imaginer()-b.imaginer());
}

// (a + bi)(c + di) = (ac−bd) + (bc+ad)i
Kompleks operator* (const Kompleks& a, const Kompleks& b) {
	Kompleks c;
	c.real = a.real*b.real-a.imaginer*b.imaginer;
	c.imaginer = a.imaginer*b.real+a.real*b.imaginer;
	return c;
	//turn Kompleks::Kompleks(a.real*b.real-a.imaginer()*b.imaginer(), a.imaginer()*b.real()+a.real()*b.imaginer());
}

// (a + bi)(c) = (ac) + (bc)i
Kompleks operator* (const Kompleks& a, const int b) {
	Kompleks c;
	c.real = a.real*b;
	c.imaginer = a.imaginer*b;
	return c;
	//turn Kompleks::Kompleks(a.real*b), a.real*b);
}
	
Kompleks operator* (const int b, const Kompleks& a) {
	Kompleks c;
	c.real = a.real*b;
	c.imaginer = a.imaginer*b;
	return c;	
	//turn Kompleks::Kompleks(a.real*b), a.real*b);
}

int Kompleks::CountKompleksInstance() {
	return n_kompleks;
}

void Kompleks::Print() {
	printf("%d", this->real);
    if (this->imaginer > 0) {
	    printf("+");
    }
	printf("%d", this->imaginer);
	printf("i\n");
}

File: Pemrograman Berbasis Objek/Responsi 4/1/Kompleks.java
public class Kompleks {
    private static int n_kompleks = 0;
    private int real;
    private int imaginer;

    public Kompleks() {
        this.real = 0;
        this.imaginer = 0;
        n_kompleks++;
    }

    public Kompleks(int _real, int _imaginer) {
        this.real = _real;
        this.imaginer = _imaginer;
        n_kompleks++;
    }

    public int getReal() {
        return this.real;
    }

    public int getImaginer(){
        return this.imaginer;        
    }

    public void setReal(int _real) {
        this.real = _real;
    }

    public void setImaginer(int _imaginer) {
        this.imaginer = _imaginer;
    }

    public Kompleks plus (Kompleks other) {
        int re = this.real + other.real;
        int im = this.imaginer + other.imaginer;
        return new Kompleks(re, im);
    }

    public Kompleks minus (Kompleks other) {
        int re = this.real - other.real;
        int im = this.imaginer - other.imaginer;
        return new Kompleks(re, im);
    }

    public Kompleks multiply (Kompleks other) {
        int re = (this.real * other.real) - (this.imaginer * other.imaginer);
        int im = (this.imaginer * other.real) + (this.real * other.imaginer);
        return new Kompleks(re, im);
    }

    public Kompleks multiply (int x) {
        int re = this.real * x;
        int im = this.imaginer * x;
        return new Kompleks(re, im);
    }

    public static int countKompleksInstance() {
        return n_kompleks;
    }

    public void print() {
        if (this.real != 0 || this.imaginer == 0) {
            System.out.print(this.real);
        } if (this.real != 0 && this.imaginer > 0) {
            System.out.print("+");
        } if (this.imaginer != 0) {
            if (this.imaginer == 1) {
                System.out.println("i");
            } else if (this.imaginer == -1) {
                System.out.println("-i");
            } else {
                System.out.println(this.imaginer + "i");
            }
        } else {
            System.out.println();
        }
    }
}